### ISSUE TITLE
TFBertMainLayer cannot be loaded from .h5

### ISSUE THREAD

**Issue by prsatyal** [tensorflow/tensorflow#58444]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Others

### Source

source

### Tensorflow Version

tf2.9

### Custom Code

No

### OS Platform and Distribution

Windows

### Current Behaviour?

```shell
I cannot load my saved model. I trained and saved bert and distilbert model but it gives me this issue, could you please help me fix this? I will also include link to the code file for your reference. 

please show how to load this model. I have trained the model for countless hours.
```

### Standalone code to reproduce the issue

```shell
https://github.com/prsatyal/issue_with_bert/blob/main/distilbert_third_try.ipynb
```

### Relevant log output

```shell
ValueError: Unknown layer: TFDistilBertModel. Please ensure this object is passed to the `custom_objects` argument. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details
```
</details>

**Comment by mohantym** [comment]
Hi @prsatyal !

Can you add the custom layer on TFDistilBertModel while loading the model and see if it works.
`loaded_model = tf.keras.models.load_model('bert_model.h5',custom_objects={'TFDistilBertModel':TFDistilBertModel})
`

[Reference.](https://github.com/tensorflow/tensorflow/issues/43721#issuecomment-1247898978)

Please post this issue on [keras-team/keras repo](https://github.com/keras-team/keras/issues) for further assistance.
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thank you!

**Comment by prsatyal** [comment]
Thank you for your help! The model loaded perfectly but is not performing as well as it did when i didnt load the model. It also gave me this error and im getting mixed answers all over the internet. How do i resolve this error and what does it mean?

WARNING:tensorflow:Error in loading the saved optimizer state. As a result, your model is starting with a freshly initialized optimizer.

**Comment by mohantym** [comment]
@prsatyal !

It might be getting resolved in 2.10 and nightly version then.
[Ref.](https://github.com/keras-team/autokeras/issues/1162)
Could you post on Keras Repo and close this thread here.
Thank you!
